docs/lg-arc4.md

```diff
@@ -1,225 +0,0 @@
-
-### Structs
-
-**Type:** `algopy.arc4.Struct`  
-**Encoding:** See [ARC4 Container Packing](#ARC4-Container-Packing)  
-**Native equivalent:** `typing.NamedTuple`
-
-ARC4 Structs are named tuples. The class keyword `frozen` can be used to indicate if a struct can be mutated. 
-Items can be accessed and mutated via names instead of indexes. Structs do not have a `.native` property,
-but a NamedTuple can be used in ABI methods are will be encoded/decode to an ARC4 struct automatically.
-
-```python
-import typing
-
-from algopy import arc4
-
-Decimal: typing.TypeAlias = arc4.UFixedNxM[typing.Literal[64], typing.Literal[9]]
-
-class Vector(arc4.Struct, kw_only=True, frozen=True):
-    x: Decimal
-    y: Decimal
-```
-
-### ARC4 Container Packing
-
-ARC4 encoding rules are detailed explicitly in the [ARC](https://github.com/algorandfoundation/ARCs/blob/main/ARCs/arc-0004.md#encoding-rules). A summary is included here.
-
-Containers are composed of a head and tail portion. 
- - For dynamic arrays, the head is prefixed with the length of the array encoded as a 16-bit number. This prefix is not included in offset calculation
- - For fixed sized items (eg. Bool, UIntN, or a StaticArray of UIntN), the item is included in the head
- - Consecutive Bool items are compressed into the minimum number of whole bytes possible by using a single bit to represent each Bool
- - For variable sized items (eg. DynamicArray, String etc), a pointer is included to the head and the data is added to the tail. This pointer represents the offset from the start of the head to the start of the item data in the tail.
-
-
-### Reference types
-
-**Types:** `algopy.Account`, `algopy.Application`, `algopy.Asset`, `algopy.gtxn.PaymentTransaction`, `algopy.gtxn.KeyRegistrationTransaction`, `algopy.gtxn.AssetConfigTransaction`, `algopy.gtxn.AssetTransferTransaction`, `algopy.gtxn.AssetFreezeTransaction`, `algopy.gtxn.ApplicationCallTransaction` 
-
-The ARC4 specification allows for using a number of [reference types](https://github.com/algorandfoundation/ARCs/blob/main/ARCs/arc-0004.md#reference-types) in an ABI method signature where this reference type refers to...
- - another transaction in the group
- - an account in the accounts array (`apat` property of the transaction)
- - an asset in the foreign assets array (`apas` property of the transaction)
- - an application in the foreign apps array (`apfa` property of the transaction)
-
-These types can only be used as parameters, and not as return types.
-
-```python
-from algopy import (
-    Account,
-    Application,
-    ARC4Contract,
-    Asset,
-    arc4,
-    gtxn,
-)
-
-class Reference(ARC4Contract):
-    @arc4.abimethod
-    def with_transactions(
-        self,
-        asset: Asset,
-        pay: gtxn.PaymentTransaction,
-        account: Account,
-        app: Application,
-        axfr: gtxn.AssetTransferTransaction        
-    ) -> None:
-        ...
-
-```
-### Mutability
```

Comment:
Not all of the documentation on this page seems to have been replicated, e.g. this section (unless I'm missing something?)